### PR TITLE
Modernize

### PR DESCRIPTION
**Pull request by rmcgibbo**
This fixes the conda-build 1.18 issue, and also updates the packages to use the new skip feature included in conda-build, which doesn't require a separate `skip.yaml` file. I also updated some packages to their latest versions.

### CODE REVIEW

**Comment by jchodera** [comment]
Niiiice!

**Comment by franknoe** [comment]
great work Robert, thanks!

Am 10/10/15 um 02:42 schrieb Robert T. McGibbon:
>
> This fixes the conda-build 1.18 issue, and also updates the packages 
> to use the new skip feature included in conda-build, which doesn't 
> require a separate |skip.yaml| file. I also updated some packages to 
> their latest versions.
>
> ------------------------------------------------------------------------
>
>
>         You can view, comment on, or merge this pull request online at:
>
> https://github.com/omnia-md/conda-recipes/pull/343
>
>
>         Commit Summary
>
>   * Use the new skip mechanism
>   * Remove skip_packages code from conda-build-all
>   * add cvxcanon required for cvxpy
>   * Update numdifftools
>
>
>         File Changes
>
>   * *M* alchemy/meta.yaml
>     <https://github.com/omnia-md/conda-recipes/pull/343/files#diff-0> (3)
>   * *D* alchemy/skip.yaml
>     <https://github.com/omnia-md/conda-recipes/pull/343/files#diff-1> (2)
>   * *M* algopy/meta.yaml
>     <https://github.com/omnia-md/conda-recipes/pull/343/files#diff-2> (9)
>   * *M* bhmm/meta.yaml
>     <https://github.com/omnia-md/conda-recipes/pull/343/files#diff-3> (4)
>   * *M* boost/meta.yaml
>     <https://github.com/omnia-md/conda-recipes/pull/343/files#diff-4> (4)
>   * *D* boost/skip.yaml
>     <https://github.com/omnia-md/conda-recipes/pull/343/files#diff-5> (2)
>   * *M* cclib/meta.yaml
>     <https://github.com/omnia-md/conda-recipes/pull/343/files#diff-6>
>     (33)
>   * *M* conda-build-all
>     <https://github.com/omnia-md/conda-recipes/pull/343/files#diff-7>
>     (38)
>   * *M* covar/meta.yaml
>     <https://github.com/omnia-md/conda-recipes/pull/343/files#diff-8> (1)
>   * *A* cvxcanon/bld.bat
>     <https://github.com/omnia-md/conda-recipes/pull/343/files#diff-9> (2)
>   * *A* cvxcanon/build.sh
>     <https://github.com/omnia-md/conda-recipes/pull/343/files#diff-10>
>     (2)
>   * *A* cvxcanon/meta.yaml
>     <https://github.com/omnia-md/conda-recipes/pull/343/files#diff-11>
>     (30)
>   * *M* cvxpy/meta.yaml
>     <https://github.com/omnia-md/conda-recipes/pull/343/files#diff-12>
>     (10)
>   * *M* doxygen/meta.yaml
>     <https://github.com/omnia-md/conda-recipes/pull/343/files#diff-13>
>     (4)
>   * *D* doxygen/skip.yaml
>     <https://github.com/omnia-md/conda-recipes/pull/343/files#diff-14>
>     (3)
>   * *M* ecos/meta.yaml
>     <https://github.com/omnia-md/conda-recipes/pull/343/files#diff-15>
>     (16)
>   * *A* ecos/run_test.py
>     <https://github.com/omnia-md/conda-recipes/pull/343/files#diff-16>
>     (33)
>   * *M* ensembler/meta.yaml
>     <https://github.com/omnia-md/conda-recipes/pull/343/files#diff-17>
>     (2)
>   * *D* ensembler/skip.yaml
>     <https://github.com/omnia-md/conda-recipes/pull/343/files#diff-18>
>     (4)
>   * *M* fastcluster/meta.yaml
>     <https://github.com/omnia-md/conda-recipes/pull/343/files#diff-19>
>     (12)
>   * *M* freeglut/meta.yaml
>     <https://github.com/omnia-md/conda-recipes/pull/343/files#diff-20>
>     (4)
>   * *D* freeglut/skip.yaml
>     <https://github.com/omnia-md/conda-recipes/pull/343/files#diff-21>
>     (3)
>   * *M* glew/meta.yaml
>     <https://github.com/omnia-md/conda-recipes/pull/343/files#diff-22>
>     (10)
>   * *D* glew/skip.yaml
>     <https://github.com/omnia-md/conda-recipes/pull/343/files#diff-23>
>     (3)
>   * *M* jom/meta.yaml
>     <https://github.com/omnia-md/conda-recipes/pull/343/files#diff-24>
>     (4)
>   * *D* jom/skip.yaml
>     <https://github.com/omnia-md/conda-recipes/pull/343/files#diff-25>
>     (3)
>   * *D* jpype1/bld.bat
>     <https://github.com/omnia-md/conda-recipes/pull/343/files#diff-26>
>     (2)
>   * *D* jpype1/build.sh
>     <https://github.com/omnia-md/conda-recipes/pull/343/files#diff-27>
>     (21)
>   * *D* jpype1/meta.yaml
>     <https://github.com/omnia-md/conda-recipes/pull/343/files#diff-28>
>     (26)
>   * *M* mdtraj/meta.yaml
>     <https://github.com/omnia-md/conda-recipes/pull/343/files#diff-29>
>     (4)
>   * *M* mingwpy/meta.yaml
>     <https://github.com/omnia-md/conda-recipes/pull/343/files#diff-30>
>     (4)
>   * *D* mingwpy/skip.yaml
>     <https://github.com/omnia-md/conda-recipes/pull/343/files#diff-31>
>     (3)
>   * *M* msmbuilder/meta.yaml
>     <https://github.com/omnia-md/conda-recipes/pull/343/files#diff-32>
>     (4)
>   * *M* numdifftools/meta.yaml
>     <https://github.com/omnia-md/conda-recipes/pull/343/files#diff-33>
>     (8)
>   * *M* openmm/meta.yaml
>     <https://github.com/omnia-md/conda-recipes/pull/343/files#diff-34>
>     (2)
>   * *D* openmm/skip.yaml
>     <https://github.com/omnia-md/conda-recipes/pull/343/files#diff-35>
>     (3)
>   * *M* openmmtools/meta.yaml
>     <https://github.com/omnia-md/conda-recipes/pull/343/files#diff-36>
>     (4)
>   * *D* openmmtools/skip.yaml
>     <https://github.com/omnia-md/conda-recipes/pull/343/files#diff-37>
>     (3)
>   * *M* openmoltools/meta.yaml
>     <https://github.com/omnia-md/conda-recipes/pull/343/files#diff-38>
>     (2)
>   * *D* openmoltools/skip.yaml
>     <https://github.com/omnia-md/conda-recipes/pull/343/files#diff-39>
>     (3)
>   * *M* pdbfixer/meta.yaml
>     <https://github.com/omnia-md/conda-recipes/pull/343/files#diff-40>
>     (2)
>   * *D* pdbfixer/skip.yaml
>     <https://github.com/omnia-md/conda-recipes/pull/343/files#diff-41>
>     (3)
>   * *M* pyhmc/meta.yaml
>     <https://github.com/omnia-md/conda-recipes/pull/343/files#diff-42>
>     (4)
>   * *M* pymbar/meta.yaml
>     <https://github.com/omnia-md/conda-recipes/pull/343/files#diff-43>
>     (6)
>   * *D* pymbar/skip.yaml
>     <https://github.com/omnia-md/conda-recipes/pull/343/files#diff-44>
>     (2)
>   * *M* pymol/meta.yaml
>     <https://github.com/omnia-md/conda-recipes/pull/343/files#diff-45>
>     (11)
>   * *D* pymol/skip.yaml
>     <https://github.com/omnia-md/conda-recipes/pull/343/files#diff-46>
>     (3)
>   * *M* python-coveralls/meta.yaml
>     <https://github.com/omnia-md/conda-recipes/pull/343/files#diff-47>
>     (31)
>   * *D* python-coveralls/skip.yaml
>     <https://github.com/omnia-md/conda-recipes/pull/343/files#diff-48>
>     (2)
>   * *M* pytram/meta.yaml
>     <https://github.com/omnia-md/conda-recipes/pull/343/files#diff-49>
>     (16)
>   * *D* pytram/skip.yaml
>     <https://github.com/omnia-md/conda-recipes/pull/343/files#diff-50>
>     (2)
>   * *M* pyublas/meta.yaml
>     <https://github.com/omnia-md/conda-recipes/pull/343/files#diff-51>
>     (6)
>   * *D* pyublas/skip.yaml
>     <https://github.com/omnia-md/conda-recipes/pull/343/files#diff-52>
>     (2)
>   * *M* pyvisfile/meta.yaml
>     <https://github.com/omnia-md/conda-recipes/pull/343/files#diff-53>
>     (8)
>   * *D* pyvisfile/skip.yaml
>     <https://github.com/omnia-md/conda-recipes/pull/343/files#diff-54>
>     (2)
>   * *M* quadprog/meta.yaml
>     <https://github.com/omnia-md/conda-recipes/pull/343/files#diff-55>
>     (3)
>   * *M* repex/meta.yaml
>     <https://github.com/omnia-md/conda-recipes/pull/343/files#diff-56>
>     (10)
>   * *D* repex/skip.yaml
>     <https://github.com/omnia-md/conda-recipes/pull/343/files#diff-57>
>     (3)
>   * *M* schema/meta.yaml
>     <https://github.com/omnia-md/conda-recipes/pull/343/files#diff-58>
>     (37)
>   * *D* schema/skip.yaml
>     <https://github.com/omnia-md/conda-recipes/pull/343/files#diff-59>
>     (2)
>   * *M* scs/build.sh
>     <https://github.com/omnia-md/conda-recipes/pull/343/files#diff-60>
>     (5)
>   * *D* scs/clapack.patch
>     <https://github.com/omnia-md/conda-recipes/pull/343/files#diff-61>
>     (19)
>   * *M* scs/meta.yaml
>     <https://github.com/omnia-md/conda-recipes/pull/343/files#diff-62>
>     (21)
>   * *A* scs/run_test.py
>     <https://github.com/omnia-md/conda-recipes/pull/343/files#diff-63>
>     (68)
>   * *M* sh/meta.yaml
>     <https://github.com/omnia-md/conda-recipes/pull/343/files#diff-64>
>     (23)
>   * *D* sh/skip.yaml
>     <https://github.com/omnia-md/conda-recipes/pull/343/files#diff-65>
>     (2)
>   * *M* silo/meta.yaml
>     <https://github.com/omnia-md/conda-recipes/pull/343/files#diff-66>
>     (4)
>   * *D* silo/skip.yaml
>     <https://github.com/omnia-md/conda-recipes/pull/343/files#diff-67>
>     (2)
>   * *M* subprocess32/meta.yaml
>     <https://github.com/omnia-md/conda-recipes/pull/343/files#diff-68>
>     (4)
>   * *D* subprocess32/skip.yaml
>     <https://github.com/omnia-md/conda-recipes/pull/343/files#diff-69>
>     (4)
>   * *M* swig/meta.yaml
>     <https://github.com/omnia-md/conda-recipes/pull/343/files#diff-70>
>     (6)
>   * *D* swig/skip.yaml
>     <https://github.com/omnia-md/conda-recipes/pull/343/files#diff-71>
>     (3)
>   * *M* yank/meta.yaml
>     <https://github.com/omnia-md/conda-recipes/pull/343/files#diff-72>
>     (4)
>   * *D* yank/skip.yaml
>     <https://github.com/omnia-md/conda-recipes/pull/343/files#diff-73>
>     (2)
>
>
>         Patch Links:
>
>   * https://github.com/omnia-md/conda-recipes/pull/343.patch
>   * https://github.com/omnia-md/conda-recipes/pull/343.diff
>
> —
> Reply to this email directly or view it on GitHub 
> <https://github.com/omnia-md/conda-recipes/pull/343>.
>


-- 

----------------------------------------------
Prof. Dr. Frank Noe
Head of Computational Molecular Biology group
Freie Universitaet Berlin

Phone: (+49) (0)30 838 75354
Web:   research.franknoe.de
Mail:  Arnimallee 6, 14195 Berlin, Germany
----------------------------------------------

File: ecos/meta.yaml [hunk 1]
```diff
@@ -1,27 +1,29 @@
 package:
   name: ecos
-  version: !!str 1.1.1
+  version: "2.0.4"
```
Comment (mpharrigan) [inline ecos/meta.yaml]: What's the difference between quotes and !!str

File: doxygen/meta.yaml [hunk 2]
```diff
@@ -2,6 +2,10 @@ package:
   name: doxygen
   version: 3.0.7
 
+build:
+    skip:
+      - [linux32 or linux64 or osx]
```
Comment (mpharrigan) [inline doxygen/meta.yaml]: below you do `[unix]` for these

File: alchemy/meta.yaml [hunk 3]
```diff
@@ -6,9 +6,12 @@ source:
   git_url: https://github.com/choderalab/alchemy.git
   git_tag: 1.0.0
 
+
 build:
   preserve_egg_dir: True
   number: 0
+  skip:
+     - [py33 or py34]
```
Comment (mpharrigan) [inline alchemy/meta.yaml]: below you do `[py3k]` (important for when we do py35?)

File: ecos/meta.yaml (hunk 1 above)
Comment (rmcgibbo) [inline ecos/meta.yaml]: They're the same.

Comment (mpharrigan) [inline ecos/meta.yaml]: should be start just using quotes. It looks a big nicer than !!str which seems aggressive

:angry: :angry: string

**Comment by jchodera** [comment]
Thanks again!

**Pull request merged**